src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -78,14 +83,12 @@ async function syncClient(
   isRetry = false,
 ) {
   Log.debug(pkg.metadata, `Processing client request: ${clientReq.clientId}`);
+  // Not including the CR data in the ref because Keycloak client IDs must be unique already
+  const name = `sso-client-${clientReq.clientId}`;
+  let client: Client;
 
   try {
-    // Not including the CR data in the ref because Keycloak client IDs must be unique already
-    const name = `sso-client-${clientReq.clientId}`;
     const token = Store.getItem(name);
-
-    let client: Client;
-
     // If an existing client is found, update it
     if (token && !isRetry) {
```

Comment:
note: need to check usecase for retry here and confirm that we not be using the token on a retry.

Reply:
Removed retries - may need to document the current behavior as it is a slight shift.

Reply:
Also related I have added a finalizer issue to handle one specific edge case here - https://github.com/defenseunicorns/uds-core/issues/523